test/RewardsStreamerMP.t.sol

```diff
@@ -2915,301 +2953,385 @@ contract CompoundTest is RewardsStreamerMPTest {
+        // assertEq(streamer.rewardIndex(), p.rewardIndex, "wrong reward index");
+    }
+
+    function check(string memory text, CheckVaultParams storage p) internal view {
+        // assertEq(rewardToken.balanceOf(p.account), p.rewardBalance, "wrong account reward balance");
+
+        RewardsStreamerMP.VaultData memory vaultData = streamer.getVault(p.account);
+
+        assertEq(
+            vaultData.stakedBalance, p.stakedBalance, string(abi.encodePacked(text, "wrong account staked balance"))
+        );
+        assertEq(
+            stakingToken.balanceOf(p.account), p.vaultBalance, string(abi.encodePacked(text, "wrong vault balance"))
+        );
+        // assertEq(vaultData.accountRewardIndex, p.rewardIndex, "wrong account reward index");
+        assertEq(vaultData.mpStaked, p.mpStaked, string(abi.encodePacked(text, "wrong account MP staked")));
+        assertEq(vaultData.mpAccrued, p.mpAccrued, string(abi.encodePacked(text, "wrong account MP accrued")));
+        assertEq(vaultData.maxMP, p.maxMP, string(abi.encodePacked(text, "wrong account max MP")));
+        assertEq(
+            vaultData.rewardsAccrued, p.rewardsAccrued, string(abi.encodePacked(text, "wrong account rewards accrued"))
+        );
+        assertEq(
+            vaultData.lockUntil,
+            expectedVaultLockState[p.account].lockEnd,
+            string(abi.encodePacked(text, "wrong account lock end"))
+        );
+    }
+
+    function _stake(address account, uint256 amount, uint256 lockPeriod, bytes4 _expectedRevert) public override {
+        stakingToken.mint(account, amount);
+        vm.prank(account);
+        stakingToken.approve(vaults[account], amount);
+        super._stake(account, amount, lockPeriod, _expectedRevert);
+        expectedRevert = FuzzTests__UndefinedError.selector;
+    }
+
     function _stake(address account, uint256 amount, uint256 lockPeriod) public override {
         stakingToken.mint(account, amount);
         vm.prank(account);
         stakingToken.approve(vaults[account], amount);
         super._stake(account, amount, lockPeriod);
     }
 
-    function testFuzz_Stake(uint256 stakeAmount, uint256 lockUpPeriod) public {
-        vm.assume(stakeAmount > 0 && stakeAmount <= MAX_BALANCE);
-        vm.assume(lockUpPeriod == 0 || (lockUpPeriod >= MIN_LOCKUP_PERIOD && lockUpPeriod <= MAX_LOCKUP_PERIOD));
-        uint256 expectedBonusMP = _bonusMP(stakeAmount, lockUpPeriod);
-        uint256 expectedMaxTotalMP = _maxTotalMP(stakeAmount, lockUpPeriod);
+    function _lock(address account, uint256 lockPeriod, bytes4 _expectedRevert) internal {
+        StakeVault vault = StakeVault(vaults[account]);
+        vm.prank(account);
+        if (_expectedRevert != NO_REVERT) {
+            vm.expectRevert(_expectedRevert);
+        }
+        vault.lock(lockPeriod);
+        expectedRevert = FuzzTests__UndefinedError.selector;
+    }
```

Comment:
Would it make sense to have this one live in `RewardsStreamerMPTest` similar to the overloaded `_stake()` function you've added there?